modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/Sidebar.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import classNames from 'classnames';
+import React, {useContext, useEffect, useState} from 'react';
+
+const SidebarContext = React.createContext();
+
+const noop = () => {};
+
+const SidebarBody = ({children}) => {
+	return <div className="sidebar-body">{children}</div>;
+};
+
+const SidebarHeader = ({children, subtitle, title}) => {
+	const {onClose} = useContext(SidebarContext);
+
+	return (
+		<div className="sidebar-header">
+			<div className="autofit-row sidebar-section">
+				<div className="autofit-col autofit-col-expand">
+					<div className="component-title">
+						<span className="text-truncate-inline">{title}</span>
+					</div>
+
+					<p className="component-subtitle">{subtitle}</p>
+
+					{children}
+				</div>
+
+				<div className="autofit-col">
+					<ClayButtonWithIcon
+						displayType="unstyled"
+						onClick={onClose}
+						small={true}
+						symbol="times"
+					/>
+				</div>
+			</div>
+		</div>
+	);
+};
+
+const Sidebar = ({children, onClose = noop, open = true}) => {
+	const [isOpen, setIsOpen] = useState(false);
+
+	// Wait until the component is rendered to show it so animation happens
```

Comment:
nit: Little bit confusing to see `500ms` [here](https://github.com/liferay-frontend/liferay-portal/pull/142/commits/2721c92e4e21b1a2b454b4e9674ca4173ce73a90#diff-9d3c1e26ed8833d6e5004ba91650afc9R58) but waiting for `100ms` [here](https://github.com/liferay-frontend/liferay-portal/pull/142/commits/2721c92e4e21b1a2b454b4e9674ca4173ce73a90#diff-414615f30f54da75ebde48fadfa94537R63).

Maybe on a later PR we might wanna try using `transitionend`?

Reply:
It's because we're not waiting for the animation to end but for the component to be rendered. I couldn't find another way to do it, 'cause first we want the component to be rendered hidden, and then open it to fire the animation. If I render it opening from the very beginning animation is not happening.

Reply:
Have you tried [useLayoutEffect](https://reactjs.org/docs/hooks-reference.html#uselayouteffect)?

> If you’re migrating code from a class component, note useLayoutEffect fires in the same phase as componentDidMount and componentDidUpdate. However, we recommend starting with useEffect first and only trying useLayoutEffect if that causes a problem.

Reply:
I will try! Thx!

Reply:
@brunobasto I tried with no success. As docs says it should be the same as use `useEffect`, that's what I did and I needed to add the `setTimeout` 'cause animation wasn't happening. I tried with `useLayoutEffect` and it's the same :(
```
	useLayoutEffect(() => {
		setIsOpen(open)
	}, [])
```

Reply:
![tenor-17147289](https://user-images.githubusercontent.com/156388/88213104-ab234a00-cc2e-11ea-8006-eaf10e493028.gif)